plugins/native/module/vtkQuakeMDLReader.cxx

```diff
@@ -0,0 +1,161 @@
+#include "vtkQuakeMDLReader.h"
+
+#include <vtkCellArray.h>
+#include <vtkCellData.h>
+#include <vtkCommand.h>
+#include <vtkDemandDrivenPipeline.h>
+#include <vtkFloatArray.h>
+#include <vtkIdTypeArray.h>
+#include <vtkInformation.h>
+#include <vtkInformationVector.h>
+#include <vtkNew.h>
+#include <vtkPointData.h>
+#include <vtkPoints.h>
+#include <vtkPolyData.h>
+#include <vtkUnsignedCharArray.h>
+
+//----------------------------------------------------------------------------
+vtkStandardNewMacro(vtkQuakeMDLReader);
+
+//----------------------------------------------------------------------------
+vtkQuakeMDLReader::vtkQuakeMDLReader()
+{
+  this->SetNumberOfInputPorts(0);
+}
+
+//----------------------------------------------------------------------------
+int vtkQuakeMDLReader::RequestData(
+  vtkInformation*, vtkInformationVector**, vtkInformationVector* outputVector)
+{
+  vtkPolyData* output = vtkPolyData::GetData(outputVector);
+
+  std::ifstream inputStream(this->FileName, std::ios::binary);
+
+  std::vector<unsigned char> buffer(std::istreambuf_iterator<char>(inputStream), {});
+
+  //??? What's a “splat”???
```

Comment:
In  `vtkF3DSplatReader`, the splats are actually just points with extra information (radius, color, etc...) because they will be drawn as ellipsoids later by F3D.  
In your case, you just want to have points and triangles.

Reply:
Alright, so does that means that I can remove all `SetNumberOfTuples()` safely?

Reply:
Please, take a look at my other comment, you can just remove all the VTK buffers you have created for now.